sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/DetectLanguageResult.cs

```diff
@@ -20,14 +20,15 @@ internal DetectLanguageResult(string id, TextDocumentStatistics statistics, ILis
         internal DetectLanguageResult(string id, string errorMessage)
             : base(id, errorMessage)
         {
+            DetectedLanguages = new ReadOnlyCollection<DetectedLanguage>(new List<DetectedLanguage>());
```

Comment:
Given you're returning a read-only list anyway, you might optimize this by having a single, lazy-initialized empty read-only collection. Would save on allocations if/when needed.

Reply:
Will follow up in person on this.